src/models/generation/renewable_generation.jl

```diff
@@ -50,21 +67,32 @@ RenewableCurtailment(; name = "init",
                 installedcapacity = 0.0,
                 econ = EconRenewable()) = RenewableCurtailment(name, status, bus, installedcapacity, econ)
 
+econ(value::RenewableCurtailment) = value.econ
+
+@forward((RenewableCurtailment, :base), RenewableGenBase)
+
 struct RenewableFullDispatch <: RenewableGen
-    name::String
-    available::Bool
-    bus::Bus
-    tech::TechRenewable
+    base::RenewableGenBase
     econ::Union{EconRenewable,Nothing}
-    internal::PowerSystemInternal
 end
 
 function RenewableFullDispatch(name, available, bus, tech, econ)
-    return RenewableFullDispatch(name, available, bus, tech, econ, PowerSystemInternal())
+    return RenewableFullDispatch(RenewableGenBase(name, available, bus, tech), econ)
 end
 
-RenewableFullDispatch(; name = "init",
-                status = false,
-                bus= Bus(),
-                installedcapacity = 0.0,
-                econ = EconRenewable()) = RenewableCurtailment(name, status, bus, installedcapacity, econ)
+function RenewableFullDispatch(; name = "init",
+                               status = false,
+                               bus= Bus(),
+                               installedcapacity = 0.0,
+                               econ = EconRenewable())
+    rc = RenewableCurtailment(name, status, bus, installedcapacity, econ)
+    return RenewableFullDispatch(PowerSystems.name(rc),
```

Comment:
This fixes what I think is a bug.  If you call RenewableFullDispatch() you actually get a value that is of type RenewableCurtailment.